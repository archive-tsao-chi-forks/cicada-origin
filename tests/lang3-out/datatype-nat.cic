@datatype Nat : Type {
  zero: Nat
  succ: (prev: Nat) -> Nat
}

@show Nat
@show Nat.zero
@show Nat.succ
@show Nat.succ(Nat.zero)
