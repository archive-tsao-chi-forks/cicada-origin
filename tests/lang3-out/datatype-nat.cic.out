@datatype Nat : Type {
  zero : @datatype Nat : Type {
    zero : Nat
    succ : (prev: Nat) -> Nat
  }
  succ : (prev: @datatype Nat : Type {
    zero : Nat
    succ : (prev: Nat) -> Nat
  }) -> Nat
}
Nat.zero
Nat.succ
Nat.succ(Nat.zero)
@datatype Bool : Type {
  true : @datatype Bool : Type {
    true : Bool
    false : Bool
  }
  false : @datatype Bool : Type {
    true : Bool
    false : Bool
  }
}
Bool.true
Bool.false
Bool.true
Bool.false
Bool.true
Bool.false
Bool.true
Bool.false
Bool.true
Bool.false

